src/components/Buttons/ShareButton.tsx

```diff
@@ -0,0 +1,33 @@
+import ShareIcon from "@mui/icons-material/Share";
+import { IconButton, Tooltip } from "@mui/material";
+import toast from "react-hot-toast";
+
+interface ShareButtonProps {
+  title: string;
+}
+
+const ShareButton: React.FC<ShareButtonProps> = ({ title }) => {
+  const handleURLCopy = () => {
+    const currentUrl = window.location.href;
+
+    navigator.clipboard
+      .writeText(currentUrl)
+      .then(() => {
+        toast.success("Recipe URL copied to clipboard!");
+      })
+      .catch((error) => {
+        toast.error("Failed to copy the URL.");
+        throw new Error(error);
```

Comment:
what does the 'throw new' error does? is it really necessary in this scenario? :)